tests/test-derivation.nix

```diff
@@ -38,24 +38,39 @@ let
       buildPhase = lib.optionalString (!dontRun) (
         ''
           mkdir -p .cache/nvim
+          status=0
+          declare -A errors
         ''
-        + lib.concatStringsSep "\n" (
-          builtins.map (
-            {
-              derivation,
-              name,
-              dontRun ? false,
-            }:
-            lib.optionalString (!dontRun) ''
-              echo "Running test for ${name}"
+        + (
+          lib.concatStringsSep "\n" (
+            builtins.map (
+              {
+                derivation,
+                name,
+                dontRun ? false,
+              }:
+              lib.optionalString (!dontRun) ''
+                echo "Running test for ${name}"
 
-              output=$(HOME=$(realpath .) ${lib.getExe derivation} -mn --headless "+q" 2>&1 >/dev/null)
-              if [[ -n $output ]]; then
-                echo "ERROR: $output"
-                exit 1
-              fi
-            ''
-          ) testList
+                output=$(HOME=$(realpath .) ${lib.getExe derivation} -mn --headless "+q" 2>&1 >/dev/null)
+                if [[ -n $output ]]; then
+                  echo "  test failure"
+                  errors[${name}]=$output
```

Comment:
Would it be simpler to print `$output` here (and set `$status`) instead of collecting + printing later?

Reply:
I'd like to avoid that, as I find it easier to look at then end of the build!

Reply:
Ah, wasn't sure if the build was full of other output or fairly empty.

Would it make sense to print twice then? Once in context and again at the end?

Reply:
Not pushing for this, just trying to understand the best options

Reply:
I think you are right, in that there would be little output. I'll change that

Reply:
I'm happy either way, assuming the output is readable. Whatever you prefer :grin: